hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -545,6 +552,39 @@ null, raftGroup, createRetryPolicy(conf), tlsConfig, conf)) {
             targetPeerId, reply);
         throw new IOException(reply.getException());
       }
+    } finally {
+      // Raft peers priorities need to be reset regardless of the result
+      // of transfer leadership
+      resetPriorities(conf, raftGroup, tlsConfig);
```

Comment:
Is it a good practice to execute `resetPriorities` in `finally`? If the leadership transfer fails but `resetPriorities` succeeds, could this lead to unexpected issues? maybe we can execute   `resetPriorities` only when the leadership transfer is successful."

Reply:
Reset priorities only if setConf success.

Reply:
Thank you for the review @xichen01 @szetszwo. Yes, I agree. I have updated it based on the review patch diff.